crates/server/src/zone_handler/catalog.rs

```diff
@@ -651,6 +649,25 @@ async fn zone_transfer(
     Err(LookupError::ResponseCode(ResponseCode::ServFail))
 }
 
+/// Helper function to construct a response message with an error response code, and send it via a
+/// response handler.
+async fn send_error_response(
+    response_code: ResponseCode,
+    request: &Request,
+    response_edns: Option<Edns>,
+    mut response_handle: impl ResponseHandler,
```

Comment:
nit: it feels like having `request` as the first parameter, and then `response_code`, `response_edns`, and `response_hande` is a touch more natural. WDYT?